core/src/main/java/org/commonjava/indy/core/change/StoreContentListener.java

```diff
@@ -386,6 +408,55 @@ private void processAllPaths( final Iterable<ArtifactStore> stores, Predicate<?
         } );
     }
 
+    /**
+     * Extensive version for clean up paths. This will clean all mergable files in affected groups regardless whether
+     * the path is in the target store cache or not.
+     * @param stores
+     * @param pathFilter
+     */
+    private void processAllPathsExt( final Iterable<ArtifactStore> stores, Predicate<? super String> pathFilter )
```

Comment:
Um, is this essential to the bugfix / feature? Metadata merges are extremely expensive...or they used to be. Maybe that's not true anymore now that we have Metadata object caching on the individual stores? This seems like enabling one remote repo that contains only a few files we care about could take hours or recovery time as we re-request irrelevant metadata files via the groups that repo is in.

Are we saying here that having those metadata files in cache eliminates most of this cost?